crud/select.lua

```diff
@@ -305,16 +361,37 @@ function select_module.call(space_name, user_conditions, opts)
             break
         end
 
+        if opts.fields then
+            tuple, err = utils.unflatten(tuple, iter.space_format)
+            if err ~= nil then
+                error(string.format("Failed to unflatten next object: %s", err))
+            end
+        end
```

Comment:
Why do you perform "unflatten" if "opts.fields" specified?

Seems you need just nullify trailing merged fields.